### DISCUSSION TITLE
npm のインストールでエラーが出てしまいます

### DISCUSSION THREAD

**Discussion by SoySoySawada** [FisaCloudAlliance/JamstackEducation#4] [Q&A]
お疲れ様です。
npmのインストールを使用としているのですが、
ターミナルにnpm installと入力しても

PS C:\FISA_WORK\JamstackEducation> npm install
npm ERR! code EAI_AGAIN
npm ERR! syscall getaddrinfo
npm ERR! errno EAI_AGAIN
npm ERR! request to https://registry.npmjs.org/@xtuc/long/-/long-4.2.2.tgz failed, reason: getaddrinfo EAI_AGAIN registry.npmjs.org
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ns220513\AppData\Local\npm-cache\_logs\2021-12-10T07_07_40_206Z-debug.log

↑のようなエラーが表示されてしまいます。

よろしくおねがいします。

**Comment by s-ueno** [comment]
Visual Studio を管理者モードで起動すると、Terminalが外部パッケージをダウンロード開始することができるかもしれません。


![image](https://user-images.githubusercontent.com/8575906/145533235-0ce37fce-3333-49de-bc92-07bc46226c14.png)

**Comment by s-ueno** [comment]
![image](https://user-images.githubusercontent.com/8575906/145533396-8ecef952-ff27-4634-87a2-126436b9a44f.png)

**Reply by SoySoySawada** [reply]
管理者で実行してみましたが、
同様のエラーが表示されてしまいました。


![image](https://user-images.githubusercontent.com/95273775/145533528-da1c4662-de90-41c3-99e3-b4d1abae85ed.png)

**Reply by nabesuke** [reply]
nslookup registry.npmjs.org
↑はどうなりますか？

**Reply by SoySoySawada** [reply]
社内ネットワークから繋いでいるのでその関係かもしれません。

**Reply by nabesuke** [reply]
社内ネットワークからということもあり、DNSサーバの名前解決ができていないのかもしれません。
https://qiita.com/neko_the_shadow/items/bf3ac2a7d2a01a76ca46

**Reply by SoySoySawada** [reply]
> 

PS C:\FISA_WORK\JamstackEducation> nslookup registry.npmjs.org
サーバー:  P2ADSO5.tisnt.tis.co.jp
Address:  172.23.2.156

名前:    registry.npmjs.org
Served by:
- e.root-servers.net
          192.203.230.10

- l.root-servers.net
          199.7.83.42

- h.root-servers.net
          128.63.2.53

- b.root-servers.net


- g.root-servers.net


- k.root-servers.net


- d.root-servers.net


- f.root-servers.net


- a.root-servers.net


- j.root-servers.net

このようなメッセージが出ます。

**Reply by nabesuke** [reply]
名前解決は大丈夫そうですね

**Reply by SoySoySawada** [reply]
やはり社内ネットワーク関係なんですかね

**Reply by nabesuke** [reply]
そうですね。
ネットワーク型のセキュリティソフトを導入されている場合、セキュリティソフトで弾かれている可能性もありそうです。
